### PR TITLE
CIV-7410 update unit tests, should be 100% method coverage now

### PR DESCRIPTION
**Pull request by drummondjm**
https://tools.hmcts.net/jira/browse/CIV-7410

Unit test updates to increase coverage 
GenerateDJFormHandler
AssignCaseToUserHandler
 
**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by m-meulendijks-v1 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java [hunk 1]
```diff
@@ -280,6 +285,26 @@ void shouldAssignCaseToApplicantSolicitorOneAndRemoveCreator1v2DSUnregisteredRes
...
+                                                                 .request(CallbackRequest.builder().eventId(
+            "ASSIGN_CASE_TO_APPLICANT_SOLICITOR1").build()).build())).isEqualTo(TASK_ID);
+    }
+
+    @Test
+    void shouldReturnSpecCamundaTask_whenSpecEvent() {
+        assertThat(assignCaseToUserHandler.camundaActivityId(CallbackParamsBuilder.builder()
+                                                                 .request(CallbackRequest.builder().eventId(
+            "ASSIGN_CASE_TO_APPLICANT_SOLICITOR1_SPEC").build()).build())).isEqualTo(TASK_ID_SPEC);
+    }
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java:306]: Add a test similar to these ones that captures the exception being thrown on line 70 of AssigneCaseToUserHandler, so we get to 100% line coverage.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java [hunk 2]
```diff
@@ -280,6 +285,26 @@ void shouldAssignCaseToApplicantSolicitorOneAndRemoveCreator1v2DSUnregisteredRes
         }
     }
 
+    @Test
+    void handleEventsReturnsTheExpectedCallbackEvents() {
+        assertThat(assignCaseToUserHandler.handledEvents()).contains(ASSIGN_CASE_TO_APPLICANT_SOLICITOR1,
+                                                                     ASSIGN_CASE_TO_APPLICANT_SOLICITOR1_SPEC);
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java:291]: Make it an exact match, so if people adds new events they are forced to look at the tests.

**Review by filippo-possenti-version1 — CHANGES_REQUESTED** [review]
See comments.

Also, the AssignCaseToUserHandler class has an if statement on line 89. For that test I would like you to make it explicit everywhere whether we are assuming toggleService.isGlobalSearchEnabled() to return true or false and add the missing tests where it's clear that we're missing either one or the other.
This way we will ensure that the methods behave as intended for the various permutations.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateDJFormHandlerTest.java [hunk 3]
```diff
@@ -271,4 +274,22 @@ public void shouldNotGenerateOneForm_when1v1() {
         }
 
     }
+
+    @Test
+    void handleEventsReturnsTheExpectedCallbackEvents() {
+        assertThat(handler.handledEvents()).contains(GENERATE_DJ_FORM, GENERATE_DJ_FORM_SPEC);
+    }
```
Comment (filippo-possenti-version1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateDJFormHandlerTest.java:281]: Same as above

**Comment by filippo-possenti-version1** [comment]
Also, take the opportunity to resolve the many SonarLint warnings that this test class has.

Not strictly part of the Acceptance Criteria, but it would be a welcome improvement.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java (hunk 1 above)
Comment (filippo-possenti-version1) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/caseassignment/AssignCaseToUserHandlerTest.java:331]: Done in @ParameterizedTest. Resolving conversation.

**Review by filippo-possenti-version1 — APPROVED** [review]

**Review by m-meulendijks-v1 — APPROVED** [review]

**Comment by filippo-possenti-version1** [comment]
The build turned red on functional tests. However this PR doesn't change the code: all it does is adding unit-tests. As a result the issues on the build are separate and the build failure is unrelated to this PR.

The conditions for this PR to be "mergeable" are actually that the new unit-tests are passing and that there are the necessary approvals.

As both the above conditions are satisfied, I'll be overriding branch protections and merge this PR.

**Pull request merged**